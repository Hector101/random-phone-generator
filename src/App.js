import * as React from 'react';
import { Router, Switch, Redirect } from 'react-router-dom';
import { createBrowserHistory } from 'history';

import routes from './routes';
import RoutesComponent from './routes/RoutesComponent';

const history = createBrowserHistory();


function App() {
  return(
    <div className="w-100 h-100">
      <Router history={history}>
        <Switch>
          {routes.map((route, i) => (
            <RoutesComponent key={i} {...route} />
          ))}
          <Redirect to='/' />
        </Switch>
      </Router>
    </div>
  )
}

export default App;
