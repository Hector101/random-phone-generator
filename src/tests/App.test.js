import React from 'react';
import { shallow } from 'enzyme';
import EnzymeToJson from 'enzyme-to-json';

import App from '../App';

it('App components mounted successfully', () => {
  const tree = shallow(<App />);

  expect(EnzymeToJson(tree)).toMatchSnapshot();
});
