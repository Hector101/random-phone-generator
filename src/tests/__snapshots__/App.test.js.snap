// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App components mounted successfully 1`] = `
<div
  className="w-100 h-100"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Switch>
      <_default
        component={[Function]}
        exact={true}
        key="0"
        path="/"
      />
      <Redirect
        push={false}
        to="/"
      />
    </Switch>
  </Router>
</div>
`;
